go-controller/pkg/ovn/pods.go

```diff
@@ -110,7 +110,7 @@ func (oc *Controller) syncPods(pods []interface{}) {
 	}
 	_, err = libovsdbops.TransactAndCheck(oc.nbClient, allOps)
 	if err != nil {
-		klog.Errorf("Could not remove stale logicalPorts from switches (%+v)", err)
```

Comment:
some of the other errors above in this function I wonder if they should also be fatal. Also, this code doesnt look right: 
```
		ls := &nbdb.LogicalSwitch{}
		if lsUUID, ok := oc.lsManager.GetUUID(n.Name); !ok {  <------TROZET if the node isn't in the cache, shouldn't we still try to get the switch based on name?
			klog.Errorf("Error getting logical switch for node %s: %s", n.Name, "Switch not in logical switch cache")
			continue
		} else {
			ctx, cancel := context.WithTimeout(context.Background(), ovntypes.OVSDBTimeout)
			defer cancel()
			ls.UUID = lsUUID
			if err := oc.nbClient.Get(ctx, ls); err != nil {   <---- TROZET If this fails shouldn't we check if err != ErrNotFound; then klog fatal
				klog.Errorf("Error getting logical switch for node %d (UUID: %d) from ovn database (%v)", n.Name, ls.UUID, err)
				continue
			}
		}

```

Reply:
Could it be a special node or something we should skip? Or is it a plain bug to have a node w/out a logical switch?

Reply:
@dcbw ^^

Reply:
@trozet  pushed an update. Please let me know if that is what you have in mind.